staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc.go

```diff
@@ -349,6 +622,35 @@ func (a *Authenticator) AuthenticateToken(token string) (user.Info, bool, error)
 	return info, true, nil
 }
 
+// getClaimJWT gets a distributed claim JWT from url, using the supplied access
+// token as bearer token.  If the access token is "", the authorization header
+// will not be set.
+// TODO: Allow passing in JSON hints to the IDP.
+func getClaimJWT(client *http.Client, url, accessToken string) (string, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// TODO: Allow passing request body with configurable information.
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return "", fmt.Errorf("while calling %v: %v", url, err)
+	}
+	if accessToken != "" {
+		req.Header.Set("Authorization", fmt.Sprintf("Bearer %v", accessToken))
+	}
+	req = req.WithContext(ctx)
+	response, err := client.Do(req)
+	if err != nil {
+		return "", err
+	}
+	defer response.Body.Close()
+	responseBytes, err := ioutil.ReadAll(response.Body)
```

Comment:
We're going to want to handle non 2XX status codes more gracefully here.

Reply:
Added code to report non-2XX statuses, not sure if there is much more that we can do here.